File: FeedItForward/frontend/src/schemas/leftoverFood.ts
import { Hawker } from "./hawker";

export interface LeftoverFood {
  leftover_food_id: number;

  available: boolean;
  name: string;
  unit_of_measurement: string;
  amount: number;
  photo: string;
  time_passed: string;

  hawker_id: number;
  hawker: Hawker;
}

export interface LeftoverFoodCreate {
  name: string;
  unit_of_measurement: string;
  amount: number;
  photo: string;
  time_passed: string;

  hawker_id: number;
}

export interface LeftoverFoodUpdate {
  leftover_food_id: number;

  available: boolean;
  name: string;
  unit_of_measurement: string;
  amount: number;
  photo: string;
  time_passed: string;
}

File: FeedItForward/backend/app/schemas/leftover_food.py
from pydantic import BaseModel

from .hawker import Hawker

class LeftoverFoodBase(BaseModel):
    name: str
    unit_of_measurement: str
    amount: int
    photo: str
    time_passed: str

class LeftoverFood(LeftoverFoodBase):
    leftover_food_id: int
    available: bool

    hawker_id: int
    hawker: Hawker

class LeftoverFoodCreate(LeftoverFoodBase):
    hawker_id: int

class LeftoverFoodUpdate(LeftoverFoodBase):
    leftover_food_id: int
    available: bool
